pkg/limits/service_test.go

```diff
@@ -2,500 +2,204 @@ package limits
 
 import (
 	"context"
-	"sync"
 	"testing"
 	"time"
 
 	"github.com/coder/quartz"
 	"github.com/go-kit/log"
-	"github.com/grafana/dskit/kv"
-	"github.com/grafana/dskit/ring"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/stretchr/testify/require"
 
-	"github.com/grafana/loki/v3/pkg/kafka"
 	"github.com/grafana/loki/v3/pkg/limits/proto"
 )
 
 func TestService_ExceedsLimits(t *testing.T) {
```

Comment:
These tests created their usage stores in the table, which made it much more difficult to initialize all the required fields. Instead, the usage store is initialized in the test, using the constructor function `newUsageStore`.